src/pages/home/report/ReportActionItem.js

```diff
@@ -267,92 +123,61 @@ class ReportActionItem extends Component {
-                                        draftMessage={this.props.draftMessage}
-                                        hidePopover={this.hidePopover}
-                                        isMini
-                                        showDeleteConfirmModal={this.showDeleteConfirmModal}
-                                    />
-                                </View>
                             </View>
-                        )}
-                    </Hoverable>
-                </PressableWithSecondaryInteraction>
-                <PopoverWithMeasuredContent
-                    isVisible={this.state.isPopoverVisible}
-                    onClose={this.hidePopover}
-                    onModalHide={this.contextMenuHide}
-                    anchorPosition={this.state.popoverAnchorPosition}
-                    animationIn="fadeIn"
-                    animationOutTiming={1}
-                    measureContent={this.measureContent}
-                    shouldSetModalVisibility={false}
-                    fullscreen={false}
-                >
-                    <ReportActionContextMenu
-                        isVisible
-                        reportID={this.props.reportID}
-                        reportAction={this.props.action}
-                        draftMessage={this.props.draftMessage}
-                        hidePopover={this.hidePopover}
-                        showDeleteConfirmModal={this.showDeleteConfirmModal}
-                    />
-                </PopoverWithMeasuredContent>
-                <ConfirmModal
-                    title={this.props.translate('reportActionContextMenu.deleteComment')}
-                    isVisible={this.state.isDeleteCommentConfirmModalVisible}
-                    onConfirm={this.confirmDeleteAndHideModal}
-                    onCancel={this.hideDeleteConfirmModal}
-                    prompt={this.props.translate('reportActionContextMenu.deleteConfirmation')}
-                    confirmText={this.props.translate('common.delete')}
-                    cancelText={this.props.translate('common.cancel')}
-                />
-            </>
+                            <MiniReportActionContextMenu
+                                reportID={this.props.reportID}
+                                reportAction={this.props.action}
+                                displayAsGroup={this.props.displayAsGroup}
+                                isVisible={
+                                    hovered
+                                    && !this.context.isActiveReportAction(this.props.action.reportActionID)
+                                    && !this.props.draftMessage
+
+                                }
+                                draftMessage={this.props.draftMessage}
+                                hidePopover={this.context.hideContextMenu}
+                                showDeleteConfirmModal={this.context.showDeleteConfirmModal}
+                            />
```

Comment:
I thought we are trying to remove the extra stuff from the `ReportActionItem` to make them lighter? What is preventing us from removing this as well? Or is there no benefit to removing it?

Reply:
Thoughts?

Reply:
Are you asking something to the effect of "why not move the `MiniReportActionContextMenu` up the component tree as well"? Because I think the reason is that when I did that before I experienced an undesirable "wiggle effect" as the position of the menu recalculates during scrolling.

Overall I think that the `MiniReportActionContextMenu` is really quite lean (a noop on native platforms), so probably best to just leave it here for now. I _think_ RN is smart enough to recognize that this component is a noop and keep it out of the render tree entirely

Reply:
That is what I was asking, but if this is only rendering on web then I think this is fine. I didn't delve too deeply into how we are handling that.